src/main/java/duke/tasklist/TaskManager.java

```diff
@@ -0,0 +1,235 @@
+package duke.tasklist;
+
+import duke.Time;
+import duke.generalmethods.GeneralMethods;
+import duke.exceptions.DukeException;
+import duke.ui.Ui;
+
+import java.util.ArrayList;
+
+public class TaskManager extends Ui {
+
+    public static final String INCORRECT_TASK_COMMAND = "Please specify a valid task to be added!";
+    public static final String INCORRECT_DEADLINE_COMMAND = "Please specify a valid task/deadline of completion!";
+    public static final String INCORRECT_EVENT_COMMAND = "Please specify a valid event/time of the event!";
+    public static final String INCORRECT_DONE_COMMAND = "Please specify the number of the task to be marked as done!";
+    public static final String INCORRECT_DELETE_COMMAND = "Please specify a valid number of the task to be deleted!";
+    public static final String INVALID_INDEX = "No number or Invalid number specified! Please specify the number on the list of the task you have completed!";
+
+    public TaskManager() {
+
+    }
+
+    /**
+     * Adds a task to the task list as an "Event" type task
+     *
+     * @param taskList Task type arraylist to store all the tasks entered by the user
+     * @param userInput takes input from the keyboard
+     * @param isPrintingPreviousList boolean to check whether it is printing from the stored list to determine whether
+     *                               to print the message that task has been successfully added. If true, then no
+     *                               message would be printed as task was already previously added
+     * @throws DukeException if there is an invalid time of event/task entered in the user input
+     */
+    public static void addTaskAsEvent(ArrayList<Task> taskList, String userInput, Boolean isPrintingPreviousList) throws DukeException {
+        if (userInput.contains("/at")) {
+            Task taskAdded = new Event(userInput);
+            if (!taskAdded.toString().equals("") && !taskAdded.getDeadline().equals("") && Time.isDateValid(taskAdded.getDeadline())) {
+                taskList.add(taskAdded);
+                if(!isPrintingPreviousList) {
+                    printTaskAddedConfirmation(taskAdded);
+                }
+            } else {
```

Comment:
I suppose you can simplify this logic?  3 levels of nested if is a bit too much.